virttest/qemu_vm.py

```diff
@@ -1185,7 +1185,16 @@ def add_balloon(devices, devid=None, bus=None, use_old_format=None):
             if not devices.has_option("device") or use_old_format is True:
                 devices.insert(StrDev('balloon', cmdline=" -balloon virtio"))
                 return
-            dev = QDevice("virtio-balloon-pci", parent_bus=bus)
+            machine_type = self.params.get("machine_type")
+            if "s390" in machine_type:    # For s390x platform
+                model = "virtio-balloon-ccw"
+            else:
+                model = "virtio-balloon-pci"
+            dev = QDevice(model)
+            if model == 'virtio-balloon-ccw':  # For s390x platform
+                dev.parent_bus = {'type': 'virtio-bus'}
+            else:
+                dev.parent_bus = bus
```

Comment:
``

if "s390" in machine_type:    # For s390x platform  
       model = "virtio-balloon-ccw"  
       bus = {'type': 'virtio-bus'}
else:  
       model = "virtio-balloon-pci"  
dev = QDevice(model, parent_bus=bus)  

``